src/main/java/org/openstreetmap/atlas/utilities/filters/AtlasEntityPolygonsFilter.java

```diff
@@ -0,0 +1,520 @@
+        {
+            return Collections.emptyList();
+        }
+        return polygonLists.entrySet().stream()
+                .flatMap(formatAndPolygonStrings -> formatAndPolygonStrings.getValue().stream().map(
+                        getPolygonConverterForFormat(formatAndPolygonStrings.getKey())::convert)
+                        .filter(Optional::isPresent).map(Optional::get).flatMap(List::stream))
+                .collect(Collectors.toList());
+    }
+
+    /**
+     * Returns predicate that filters out MultiPolygons that overlap with any Polygons or outer
+     * MultiPolygons in the provided QuadTree. This also adds any non-intersecting MultiPolygon to
+     * the quad tree.
+     *
+     * @param vettedPolygons
+     *            non-overlapping polygons and multipolygons
+     * @param polygonType
+     *            whether this is an include or exclude Polygon
+     * @return filter for multipolygons
+     */
+    private static Predicate<MultiPolygon> notOverlappingMultipolygon(
+            final QuadTree<Located> vettedPolygons, final Type polygonType)
+    {
+        return multiPolygon ->
+        {
+            if (vettedPolygons.get(multiPolygon.bounds()).stream()
+                    .filter(located -> located instanceof Polygon).map(located -> (Polygon) located)
+                    .noneMatch(multiPolygon::overlaps)
+                    && vettedPolygons.get(multiPolygon.bounds()).stream()
+                            .filter(located -> located instanceof MultiPolygon)
+                            .map(located -> (MultiPolygon) located).map(MultiPolygon::outers)
+                            .flatMap(Set::stream).noneMatch(multiPolygon::overlaps))
+            {
+                vettedPolygons.add(multiPolygon.bounds(), multiPolygon);
+                return true;
+            }
+            else
+            {
+                logger.warn("Dropping {} MultiPolygon {}", polygonType, multiPolygon);
+                return false;
+            }
+        };
+    }
+
+    /**
+     * Returns predicate that filters out polygons that intersect with any polygons in the provided
+     * QuadTree. This also adds any non-intersecting polygon to the quad tree.
+     *
+     * @param vettedPolygons
+     *            non-overlapping polygons and multipolygons
+     * @param polygonType
+     *            whether this is an include or exclude Polygon
+     * @return filter for polygons
+     */
+    private static Predicate<Polygon> notOverlappingPolygon(final QuadTree<Located> vettedPolygons,
+            final Type polygonType)
+    {
+        return polygon ->
+        {
+            if (vettedPolygons.get(polygon.bounds()).stream()
+                    .filter(located -> located instanceof Polygon).map(located -> (Polygon) located)
+                    .noneMatch(polygon::overlaps)
+                    && vettedPolygons.get(polygon.bounds()).stream()
+                            .filter(located -> located instanceof MultiPolygon)
```

Comment:
Same question about inners here.

Reply:
The ```Polygon::overlaps(MultiPolygon``` method handles the intersection and relationship between the `MultiPolygon```'s outers and inners